### ISSUE TITLE
Cannot read property 'jsonData' of undefined (webpack5 and storybook/html)

### ISSUE THREAD

**Issue by raduchiriac** [storybookjs/storybook#15463]
**Describe the bug**
I started a project using `npx sb init --builder webpack5` and chose **html** for my future components. I cannot edit the stories because the builder breaks.

**To Reproduce**
I start the project with `npm run storybook` and whenever I try to edit and save anything (like a **.stories.js**) the hot module restarts building but breaks immediately with an error:

```sh
webpack building...
3% setup watch run/Users/raduchiriac/Projects/-node/sb-temp/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/json/JsonGenerator.js:119
		let data = module.buildInfo.jsonData;
		                            ^

TypeError: Cannot read property 'jsonData' of undefined
```

**System**
System:
    OS: macOS Mojave 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-4980HQ CPU @ 2.80GHz
  Binaries:
    Node: 14.17.2 - ~/.nvm/versions/node/v14.17.2/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.13 - ~/.nvm/versions/node/v14.17.2/bin/npm
  Browsers:
    Chrome: 91.0.4472.114
    Firefox: 88.0
    Safari: 14.1.1
  npmPackages:
    @storybook/addon-actions: ^6.3.2 => 6.3.2
    @storybook/addon-essentials: ^6.3.2 => 6.3.2
    @storybook/addon-links: ^6.3.2 => 6.3.2
    @storybook/builder-webpack5: ^6.3.2 => 6.3.2
    @storybook/html: ^6.3.2 => 6.3.2
    @storybook/manager-webpack5: ^6.3.2 => 6.3.2

**Additional context**

**Comment by shilman** [comment]
Can you please create a reproduction by running `npx sb@next repro`, following the instructions, and linking it in your issue description? We prioritize issues with reproductions over those without. Thank you! 🙏

**Comment by raduchiriac** [comment]
My issue is very easy to replicate:

1. `npx sb init --builder webpack5 -t html`
2. `npm run storybook`
3.  edit any file you like, `webpack-hot-middleware` restarts but breaks with an error: `TypeError: Cannot read property 'jsonData' of undefined`


sorry, I get a yarn2 error about the "output folder" while doing the reproduction stript.

**Comment by raduchiriac** [comment]
> Can you please create a reproduction by running `npx sb@next repro`, following the instructions, and linking it in your issue description? We prioritize issues with reproductions over those without. Thank you! 🙏

sorry, I get a yarn2 error about the "output folder" while doing the reproduction script.

**Comment by gaganjot07** [comment]
I just updated storybook to version 6.3 and webpack to version 5 and I get the same error

![image](https://user-images.githubusercontent.com/26659052/124321075-0f42d380-db4b-11eb-96b9-812cf484aa03.png)

**Comment by shilman** [comment]
@raduchiriac in what project?

**Comment by raduchiriac** [comment]
It seems very much unrelated to storybook now. Focus moved here: https://github.com/webpack/webpack/issues/13691

**Comment by betodasilva** [comment]
Same error here
In my case, I have an Angular project and started storybook with `npx sb@next init`
Intial build works, but whenever I save some file it breaks
<img width="1074" alt="Screen Shot 2021-07-05 at 02 31 28" src="https://user-images.githubusercontent.com/17956938/124422083-1b8c8380-dd39-11eb-9de0-55f2727002a4.png">

**Comment by raduchiriac** [comment]
@betodasilva Try to use the `resolutions` solution I posted in the linked webpack issue above.

**Comment by betodasilva** [comment]
thanks @raduchiriac it worked.

to add my two cents:
I'm using yarn so first time I ran `yarn install` I got an error saying _package-lock not found_ (something like that, forgot to screenshot the log)

but found a solution, that is, edit the preinstall line to the following:
`"preinstall": "npm install --package-lock-only --ignore-scripts && npx npm-force-resolutions"
` 

now it's working fine

**Comment by jogelin** [comment]
Seems to be fixed in webpack `v5.42.1`:

![image](https://user-images.githubusercontent.com/954509/124565157-04858880-de42-11eb-8270-2223e733f09f.png)

Could be fixed by setting:

`package.json`
```
...
"resolutions": {
    "webpack": "5.42.1"
  }
```
